byron/ledger/executable-spec/src/Ledger/Core.hs

```diff
@@ -479,9 +487,15 @@ deriveGoblin ''VKeyGenesis
 instance GeneOps g => Goblin g Hash where
   tinker gen
     = tinkerRummagedOrConjureOrSave
-        ((Hash . (`mod` 30))
-           <$$> tinker ((\(Hash x) -> x) <$> gen))
-  conjure = saveInBagOfTricks =<< (Hash . (`mod` 30) <$> conjure)
+        ((Hash . Just . (`mod` 30))
+           <$$> tinker (unwrapValue <$> gen))
+    where
+      unwrapValue (Hash (Just x)) = x
+      unwrapValue (Hash Nothing) =
+        error $  "tinker Hash instance: trying to tinker with an invalid hash"
+              ++ " (which contains nothing)"
```

Comment:
I'm wondering - do we really never produce `Hash Nothing` values in hedgehog shrink-trees which are tinkered with by goblins?

All the `HasHash` instances I see are of the form `hash = Hash . Just . H.hash`, which indicates that the answer to 👆 might be yes, but I haven't looked exhaustively.

We could always have the `tinker` of `Hash Nothing` remain `Hash Nothing`, if there's concern about us `error`ing. But presumably since this merged, the testsuites ran and didn't `error`!

Mostly just curious.

Reply:
> All the HasHash instances I see are of the form hash = Hash . Just . H.hash, which indicates that the answer to point_up_2 might be yes, but I haven't looked exhaustively.

Yeah, that'd be my guess as well, but you're in a better position of answering this ;) Why would the goblins ever produce `Hash Nothing` if we don't have code that allows this?

> Mostly just curious.

+1 :slightly_smiling_face:

Reply:
> Why would the goblins ever produce Hash Nothing if we don't have code that allows this?

Given this `Goblin` instance they wouldn't, since we tinker underneath the `Just`. It just occurred to me that the original generator (which goblins will tinker with) could have generated a `Hash Nothing` and thus would trigger the error. I can look more into it 🙂

Reply:
Oh, you meant the original generator. Well, that would imply that we're generating an invalid signal, which we shouldn't. So I wouldn't expect a problem here, but if we run into one we can look further into this ;)

Reply:
Ok, we’re on the same page then, let’s not look unless it comes up 🙂